### ISSUE TITLE
Language Selection

### ISSUE THREAD

**Issue by cthibaultatx** [cityofaustin/janis#35]
Gives residents the ability to change most content to 1 of 8 human-translated languages.

- [x] Languages should always be listed in the language itself (Espanol, not Spanish)
- [x] Language selection, although a separate navigation element, persists globally
- [x] default copy = "Choose language: English" paired with down-facing caret
- [x] when a user has set a language, any link they click should be prefixed by the language code. We need to research if there is a react-router way to do this? Or how it might work without react router too.

![screen shot 2018-01-09 at 8 37 32 am](https://user-images.githubusercontent.com/20911665/34726019-642da45a-f518-11e7-9a4f-b1307c80e580.png)

# Resident Value
Austin is becoming a more diverse city, supporting residents who primarily speak English and Spanish but a growing number of individuals and families who speak Vietnamese, Chinese, Arabic, and other languages. By making content accessible in top Austin languages–our content, our services serve more people.

**Comment by mateoclarke** [comment]
This will have significant interaction with the Joplin API. @benweatherman, we need to coordinate on how we flip the language switch and request that from the API instead. I'm also thinking about text in global site components like Headers and Footers. I guess that should live as content data with translations in Joplin too? How would we add this to the Django models? Currently we are hardcoding from the mocks.

**Comment by mateoclarke** [comment]
Looks like you are already thinking about some of this in https://github.com/cityofaustin/joplin/issues/17

**Comment by mateoclarke** [comment]
This PR #46 starts to work on language selection in terms of appearance, but only partially. And it has some discussion about the bigger picture problems yet to be solved:
- the mechanics of setting a language cookie in the client for the server to access
- updating routing to follow best practice, ex: `austin.gov/es/`.

**Comment by mateoclarke** [comment]
Another issue to solve here:
- when a user has set a language, any link they click should be prefixed by the language code. We need to research if there is a react-router way to do this? Or how it might work without react router too.

**Comment by cthibaultatx** [comment]
@mateoclarke from a city staff member at Asian American Resource Center:
Hi Celine,
 
Thanks for calling us. Sorry for the late reply. Following are my personal point of view.
 
From my personal experience, I didn’t pay attention on the abbreviations. So I did some research, I noticed that they usually use simplified Chinese and Traditional Chinese. For Simplified Chinese, it is 简体中文，traditional Chinese is 繁體中文.
 
The abbreviation you mentioned is for typing Chinese on computer only. People will not use it in everyday life. If you want to tell people there has Chinese service for them, 简体中文 will be the best one for people reading simplified Chinese, and繁體中文 for people reading traditional Chinese.
 
Hope it helps! If you have any further questions, feel free to contact me.
 
Thanks,
